app/models/project.rb

```diff
@@ -0,0 +1,6 @@
+class Project < ActiveRecord::Base
+
+  validates_presence_of :name, :description, :start_date, :end_date, :user_id
```

Comment:
Organize your model settings. It is not requirement but is is very useful. See example

`class Channel < ActiveRecord::Base
  RECOMMENDED_SIZE = 5

#-------------Plugins------------------
  acts_as_followable

#-------------Associations-------------
  belongs_to :owner, class_name: 'User'
  belongs_to :position_department, class_name: 'Department'

  has_one :location, as: :owner, dependent: :destroy

  has_many :position_types, through: :position_department
  has_many :updates, as: :postable, class_name: 'Post', dependent: :destroy

#-------------Scopes-------------------
  scope :by_professions, -> (position_type_ids) { joins(:position_types).where(position_types: { id: position_type_ids }) }
  scope :by_location, -> (*location) { where(id: Location.near(*location).where(owner_type: 'Group').map(&:owner_id)) }

  accepts_nested_attributes_for :location

#-------------Callbacks----------------
  after_create :create_header
  after_save :check_moderator

#-------------Class methods------------
  class << self
    def recommended(user)
      channels = initial_recommended(user).limit(RECOMMENDED_SIZE)
      channels += fullfill_recommended(channels) if channels.size < RECOMMENDED_SIZE
      channels
    end

    private

    def fullfill_recommended(channels)
      where.not(id: channels.map(&:id)).order('RANDOM()').limit(RECOMMENDED_SIZE - channels.size)
    end

    def initial_recommended(user)
      profession_ids = user.professions.pluck(:id)

      if profession_ids.any?
        by_professions(profession_ids)
      else
        order('RANDOM()')
      end
    end
  end

#-------------Instance methods---------
  private

  def check_moderator
    owner.follow self if owner && !owner.following?(self)
  end

  def create_header
    create_logo
  end
end`